integration_tests/src/main/python/csv_test.py

```diff
@@ -268,27 +269,28 @@ def test_round_trip(spark_tmp_path, data_gen, v1_enabled_list):
             lambda spark : spark.read.schema(schema).csv(data_path),
             conf=updated_conf)
 
-@allow_non_gpu('FileSourceScanExec')
+@allow_non_gpu('org.apache.spark.sql.execution.LeafExecNode')
 @pytest.mark.parametrize('read_func', [read_csv_df, read_csv_sql])
 @pytest.mark.parametrize('disable_conf', ['spark.rapids.sql.format.csv.enabled', 'spark.rapids.sql.format.csv.read.enabled'])
 def test_csv_fallback(spark_tmp_path, read_func, disable_conf):
     data_gens =[
         StringGen('(\\w| |\t|\ud720){0,10}', nullable=False),
         byte_gen, short_gen, int_gen, long_gen, boolean_gen, date_gen]
- 
+
     gen_list = [('_c' + str(i), gen) for i, gen in enumerate(data_gens)]
     gen = StructGen(gen_list, nullable=False)
     data_path = spark_tmp_path + '/CSV_DATA'
     schema = gen.data_type
     updated_conf=_enable_all_types_conf.copy()
     updated_conf[disable_conf]='false'
- 
+
     reader = read_func(data_path, schema)
     with_cpu_session(
             lambda spark : gen_df(spark, gen).write.csv(data_path))
     assert_gpu_fallback_collect(
             lambda spark : reader(spark).select(f.col('*'), f.col('_c2') + f.col('_c3')),
-            'FileSourceScanExec',
+            # TODO add support for lists
```

Comment:
Is this TODO needed?

Reply:
I think it would be great to have this function accept a list of fallback classes with the semantics: ensure one of the provided fallback classes has been used. Especially if we want to limit the use of the superclass feature.

However, if we want to establish a culture of filing issues instead of leaving TODO comments going forward, it would make sense to me.